pkg/k8shandler/prometheus_rule.go

```diff
@@ -0,0 +1,102 @@
+package k8shandler
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/client/monitoring/v1"
+	v1alpha1 "github.com/openshift/elasticsearch-operator/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sYAML "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+const (
+	alertsFilePath = "files/prometheus_alerts.yml"
+	rulesFilePath  = "files/prometheus_rules.yml"
+)
+
+func CreateOrUpdatePrometheusRules(dpl *v1alpha1.Elasticsearch) error {
```

Comment:
I don't see a reason to have both this available at both scopes. We can likely just use `CreateOrUpdatePrometheusRules` and move the logic from within `createOrUpdatePrometheusRules` into this function.

Reply:
We don't get anything from having the private implementation though, we never call it again anywhere else. To me its just making the code more difficult to read without adding any DRY